flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java

```diff
@@ -67,12 +68,53 @@
 
 	// ------------------------------------------------------------------------
 
+	/**
+	 * Allocates a new (header and contents) buffer and adds some header information for the frame
```

Comment:
This method only allocates memory for the header and no contents (or contents is empty). Maybe we could update the JavaDoc accordingly.